api/s2n.h

```diff
@@ -1559,6 +1559,43 @@ S2N_API extern int s2n_client_hello_get_session_id_length(struct s2n_client_hell
  */
 S2N_API extern int s2n_client_hello_get_session_id(struct s2n_client_hello *ch, uint8_t *out, uint32_t *out_length, uint32_t max_length);
 
+/**
+ * Get the length of the compression methods list sent in the Client Hello.
+ *
+ * @param ch A pointer to the Client Hello
+ * @param out_length An out pointer. Will be set to the length of the compression methods list in bytes.
+ * @returns S2N_SUCCESS on success. S2N_FAILURE on failure
+ */
+S2N_API extern int s2n_client_hello_get_compression_methods_length(struct s2n_client_hello *ch, uint32_t *out_length);
+
+/**
+ * Retrieves the list of compression methods sent in the Client Hello.
+ *
+ * Use `s2n_client_hello_get_compression_methods_length()`
+ * to retrieve how much memory should be allocated for the `out` buffer in advance.
+ *
+ * @note Compression methods were removed in TLS13 and therefore the only valid value in this list is the
+ * "null" compression method when TLS13 is negotiated. 
+ * 
+ * @param ch A pointer to the Client Hello
+ * @param buffer A pointer to the buffer that s2n will write the compression methods to. This buffer MUST be the size of `buffer_length`
+ * @param buffer_length The size of `buffer`.
+ * @param out_length An out pointer. s2n will set its value to the size of the compression methods list in bytes.
+ * @returns S2N_SUCCESS on success. S2N_FAILURE on failure
+ */
+S2N_API extern int s2n_client_hello_get_compression_methods(struct s2n_client_hello *ch, uint8_t *buffer, uint32_t buffer_length, uint32_t *out_length);
+
+/**
+ * Access the Client Hello protocol version
+ *
+ * @note This field is a legacy field in TLS13 and is no longer used to negotiate the
+ * protocol version of the connection. Therefore this method should only be used for logging or fingerprinting.
```

Comment:
```suggestion
 * @note This field is a legacy field in TLS1.3 and is no longer used to negotiate the
 * protocol version of the connection. It will be set to TLS1.2 even if TLS1.3 is negotiated. Therefore this method should only be used for logging or fingerprinting.
```